src/LiveComponent/src/EventListener/AddLiveAttributesSubscriber.php

```diff
@@ -58,13 +61,22 @@ public static function getSubscribedServices(): array
             LiveComponentHydrator::class,
             UrlGeneratorInterface::class,
             Environment::class,
+            ComponentFactory::class,
             '?'.CsrfTokenManagerInterface::class,
         ];
     }
 
     private function getLiveAttributes(MountedComponent $mounted): ComponentAttributes
     {
         $name = $mounted->getName();
+
+        try {
+            /** @var ComponentMetadata $metadata */
+            $metadata = $this->container->get(ComponentFactory::class)->metadataFor($name);
+        } catch (\InvalidArgumentException $e) {
+            throw new NotFoundHttpException(sprintf('Component "%s" not found.', $name), $e);
+        }
+
```

Comment:
The metadata is [available on the event](https://github.com/symfony/ux/blob/e4ab138168b481c051cb63b80b6f20d19c5d03c3/src/LiveComponent/src/EventListener/AddLiveAttributesSubscriber.php#L38).

Reply:
:man_facepalming: thanks